src/filters/ws.rs

```diff
@@ -134,10 +134,7 @@ where
     fn into_response(self) -> Response {
         let on_upgrade = self.on_upgrade;
         let config = self.ws.config;
-        let fut = self
-            .ws
-            .body
-            .on_upgrade()
+        let fut = hyper::upgrade::on(Response::new(self.ws.body))
```

Comment:
We can merge
`Response::new(self.ws.body)` and `let mut res = http::Response::default();`  to

```
let mut res   = Response::new(self.ws.body); 
let fut = hyper::upgrade::on(&mut res)
```

Reply:
@seanmonstar Is it safe to replace `res` as suggested in the above comment? AFAICT, the difference is that `res` gets constructed from `self.ws.body`.

Reply:
I ported `ws.rs` to viz-rs. 
You can see that https://github.com/viz-rs/viz/blob/master/viz-core/src/ws.rs.

Reply:
https://github.com/hyperium/hyper/blob/master/src/upgrade.rs#L258-L287